ci/gdk_build.template.steps.yaml

```diff
@@ -28,19 +28,12 @@ env:
   FASTBUILD_BROKERAGE_PATH: "\\\\fastbuild-brokerage.${CI_ENVIRONMENT}-intinf-eu1.i8e.io\\samba"
 
 steps:
-  - label: "build-and-test-GDK-:windows:-ENGINE_COMMIT_HASH_PLACEHOLDER"
-    command: "powershell ./ci/setup-build-test-gdk.ps1 -target_platform Win64"
-    <<: *common # This folds the YAML named anchor into this step. Overrides, if any, should follow, not precede.
-    artifact_paths:
-      - "../UnrealEngine/Engine/Programs/AutomationTool/Saved/Logs/*"
-      - "../UnrealEngine/Samples/StarterContent/Saved/Logs/*"
-    env:
-      ENGINE_COMMIT_HASH: "ENGINE_COMMIT_HASH_PLACEHOLDER"
-
-  - label: "build-GDK-:linux:-ENGINE_COMMIT_HASH_PLACEHOLDER"
-    command: "powershell ./ci/setup-build-test-gdk.ps1 -target_platform Linux"
+  - label: "build-ENGINE_COMMIT_HASH_PLACEHOLDER-BUILD_PLATFORM_PLACEHOLDER-BUILD_TARGET_PLACEHOLDER-BUILD_STATE_PLACEHOLDER"
```

Comment:
If we set these as environment variables before piping them to `buildkite-agent pipeline upload` it will attempt to interpolate them. This has a few benefits. 

- Easier to read (imo)
- We don't need to deal with our own templating. 

This is done above in the fastbuild configuration.

---

For context on why I didn't do this for the NFR things, the steps I was uploading had shell-variable-like elements that should _not_ be expanded, and would actually fail when Buildkite attempted to do so, so I had to disable the interpolation and do the templating manually.

I realise that I originally pointed you toward that as an example, which was very misleading 😬

Reply:
Agreed - fixed it now 😃